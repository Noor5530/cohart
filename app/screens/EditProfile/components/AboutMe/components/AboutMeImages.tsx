import React from "react";
import { View, Image, TouchableOpacity } from "react-native";
import ProfileThumbnail from "assets/Placeholder-5.png";
import ProfileThumbnail1 from "assets/Placeholder-2.png";
import ProfileThumbnail2 from "assets/Placeholder-1.png";
import ProfileThumbnail3 from "assets/Placeholder-4.png";
import ProfileThumbnail4 from "assets/Placeholder-6.png";
import ProfileThumbnail5 from "assets/Placeholder-3.png";
import EditIcon from "assets/ic_edit.png";

import FastImage from "react-native-fast-image";
import { ActivityIndicator } from "react-native-paper";

import styles from "./style";
import { neonYellow } from "config/colors";

interface Props {
  image1: string | null;
  image2: string | null;
  image3: string | null;
  image4: string | null;
  image5: string | null;
  image6: string | null;
  pickImages(imgNum: any): () => void;
  imageLoading: string[];
}
export default function AboutMeImages(props: Props) {
  const {
    image1 = "",
    image2 = "",
    image3 = "",
    image4 = "",
    image5 = "",
    image6 = "",
    pickImages,
    imageLoading,
  } = props;

  console.log(image1)
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.mainImage}
        disabled={imageLoading.includes("1")}
        onPress={pickImages(1)}
      >
        <FastImage
          resizeMode={"cover"}
          source={image1 ? { uri: image1 } : ProfileThumbnail}
          style={styles.profileImage}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("1") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>
      <TouchableOpacity disabled={imageLoading.includes("2")} style={styles.firstImage} onPress={pickImages(2)}>
        <FastImage
          resizeMode={"cover"}
          source={image2 ? { uri: image2 } : ProfileThumbnail1}
          style={styles.featureFirstImage}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("2") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>
      <TouchableOpacity disabled={imageLoading.includes("3")} style={styles.secondImage} onPress={pickImages(3)}>
        <FastImage
          resizeMode={"cover"}
          source={image3 ? { uri: image3 } : ProfileThumbnail2}
          style={styles.featureSecondImage}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("3") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>
      <TouchableOpacity disabled={imageLoading.includes("4")} style={styles.thirdImage} onPress={pickImages(4)}>
        <FastImage
          resizeMode={"cover"}
          source={image4 ? { uri: image4 } : ProfileThumbnail3}
          style={styles.featureThird}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("4") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>
      <TouchableOpacity disabled={imageLoading.includes("5")} style={styles.forthImage} onPress={pickImages(5)}>
        <FastImage
          resizeMode={"cover"}
          source={image5 ? { uri: image5 } : ProfileThumbnail5}
          style={styles.featureForthImage}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("5") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>

      <TouchableOpacity disabled={imageLoading.includes("6")} style={styles.fifthImage} onPress={pickImages(6)}>
        <FastImage
          resizeMode={"cover"}
          source={image6 ? { uri: image6 } : ProfileThumbnail4}
          style={styles.featureFifthImage}
        />
        <View style={styles.overlay}>
          {imageLoading.includes("6") ? (
            <ActivityIndicator color={neonYellow} size="small" />
          ) : (
            <Image source={EditIcon} />
          )}
        </View>
      </TouchableOpacity>
    </View>
  );
}
