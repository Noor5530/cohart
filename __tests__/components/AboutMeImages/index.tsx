import "react-native";
import React from "react";
import AboutMeImages from "../../../app/screens/EditProfile/components/AboutMe/components/AboutMeImages";
import renderer from "react-test-renderer";

jest.useFakeTimers();
jest.mock("rn-fetch-blob", () => ({}));

jest.mock("react-redux", () => ({
  useDispatch: () => {},
  useSelector: () => ({
    user: {
      affiliations: [],
      bio: "test",
      referral_by: {
        cover_image: "hnnj",
      },
    },
  }),
}));

test("renders correctly", () => {
  const tree = renderer
    .create(
      <AboutMeImages
        image1={""}
        image2={""}
        image3={""}
        image4={""}
        image5={""}
        image6={""}
        pickImages={jest.fn}
        imageLoading={[]}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
