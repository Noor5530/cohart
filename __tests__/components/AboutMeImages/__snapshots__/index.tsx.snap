// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 200,
      "justifyContent": "center",
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 200,
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 200,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-5.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 20,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 61,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-2.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": undefined,
        "opacity": 1,
        "position": "absolute",
        "top": 40,
        "zIndex": -6,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 75,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-1.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 40,
        "left": undefined,
        "opacity": 1,
        "position": "absolute",
        "zIndex": -5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 75,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-4.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 25,
        "opacity": 1,
        "position": "absolute",
        "right": -5,
        "zIndex": -5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 75,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-3.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": undefined,
        "top": 50,
        "zIndex": -5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 75,
            "width": undefined,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/Placeholder-6.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/ic_edit.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
